File: 백준/Silver/1515. 수 이어 쓰기/수 이어 쓰기.py
import sys
input = sys.stdin.readline

def find(remain):
    n = 0
    pt = 0
    while pt<len(remain):
        n +=1
        current_n = str(n)
        for char in current_n:
            if pt<len(remain) and char == remain[pt]:
                pt +=1
            if pt == len(remain):
                return n


word = input().strip()
print(find(word))

File: 백준/Silver/1515. 수 이어 쓰기/수 이어 쓰기.java
import java.util.*;
import java.io.*;
public class Main{
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String remain =  br.readLine().trim();
        int n = 0;
        int pt = 0;
        while(true){
            n++;
            String currentN = String.valueOf(n);
            for(int i=0;i<currentN.length();i++){
                if(pt<remain.length() && currentN.charAt(i) == remain.charAt(pt)){
                    pt++;
                }
                if(pt == remain.length()){
                    System.out.println(n);
                    return;
                }
            }
        }
        }

    }
